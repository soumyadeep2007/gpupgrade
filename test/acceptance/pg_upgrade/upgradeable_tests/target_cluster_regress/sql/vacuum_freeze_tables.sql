-- Copyright (c) 2017-2021 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

-- should be able to vacuum freeze the tables
VACUUM FREEZE vf_tbl_heap;
VACUUM FREEZE vf_tbl_ao;
VACUUM FREEZE vf_tbl_aoco;

-- should be able to create a new table without any warnings related to vacuum
CREATE TABLE upgraded_vf_tbl_heap (LIKE vf_tbl_heap);
INSERT INTO upgraded_vf_tbl_heap SELECT * FROM vf_tbl_heap;
VACUUM FREEZE upgraded_vf_tbl_heap;
SELECT * FROM upgraded_vf_tbl_heap;
