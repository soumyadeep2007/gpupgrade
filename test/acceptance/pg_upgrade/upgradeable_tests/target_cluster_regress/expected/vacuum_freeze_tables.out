-- Copyright (c) 2017-2021 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Validate that the upgradeable objects are functional post-upgrade
--------------------------------------------------------------------------------

-- should be able to vacuum freeze the tables
VACUUM FREEZE vf_tbl_heap;
VACUUM
VACUUM FREEZE vf_tbl_ao;
VACUUM
VACUUM FREEZE vf_tbl_aoco;
VACUUM

-- should be able to create a new table without any warnings related to vacuum
CREATE TABLE upgraded_vf_tbl_heap (LIKE vf_tbl_heap);
CREATE
INSERT INTO upgraded_vf_tbl_heap SELECT * FROM vf_tbl_heap;
INSERT 10
VACUUM FREEZE upgraded_vf_tbl_heap;
VACUUM
SELECT * FROM upgraded_vf_tbl_heap;
 a  | b  
----+----
 1  | 1  
 2  | 2  
 3  | 3  
 4  | 4  
 5  | 5  
 6  | 6  
 7  | 7  
 8  | 8  
 9  | 9  
 10 | 10 
(10 rows)
